packages/teleport-uidl-validator/src/decoders/custom-combinators.ts

```
@@ -43,8 +43,8 @@ export class CustomCombinators<A> {
 
   static isValidElementName(): CustomCombinators<string> {
     return new CustomCombinators<string>((json: string) => {
-      const fileNameRegex = new RegExp('^[a-zA-Z]+[a-zA-Z0-9-_]*$')
-      if (json && typeof json === 'string' && fileNameRegex.test(json)) {
+      const fileNameRegex = new RegExp('^$|^[a-zA-Z]+[a-zA-Z0-9-_]*$')
+      if (typeof json === 'string' && fileNameRegex.test(json)) {
```

Well, `empty-strings` are not valid element names. So, it will be great if we can do thr string length check and throw error if it is empty strings.

---

@JayaKrishnaNamburu, means we need to throw an error for empty element name? was thinking that the empty element name is valid, and made change accordingly.

---

Yes, actually `name` field is optional. But if the field exists then it should be a valid name. So, `name` can't be a empty string.

---

I think that thing was already happening before, throwing an error if element `name` has an empty string, and it's optional.

---

Yes, the issue was to improve error message 😃